test/controllers/EditSetRecordControllerSpec.scala

```diff
@@ -1750,8 +1752,8 @@ class EditSetRecordControllerSpec extends BaseControllerSpec {
     */
   private def givenCreatorsExist(returnedCreators: Seq[Creator] = Seq.empty)(implicit
     referenceDataService: ReferenceDataService
-  ): ScalaOngoingStubbing[Seq[Creator]] =
-    when(referenceDataService.getCreators).thenReturn(returnedCreators)
+  ): ScalaOngoingStubbing[IO[Seq[Creator]]] =
```

Comment:
Could we return a non-empty list of creators, similarly to what we do with legal statuses and places of deposit?

Reply:
I've tried changing this to a list of creators, but mockito doesn't like it:
```
[info]  - this invocation of 'apply' method:
[info]     editSetRecordEdit.apply(
[info]     User("1234"),
[info]     "COAL 80 Sample",
[info]     "edit-set.record.edit.title",
[info]     EditSetRecord("COAL 80/80/8", "COAL.2022.V8RJW.P", "Bedlington Colliery, Newcastle Upon Tyne. Photograph depicting: view of pithead baths. (H)", "1969", "", "", "1", "1", "1969", "31", "12", "1969", "ref.1", "1", "", "", "", List(), List(), List("48N", "46F", "8R6")),
[info]     List(LegalStatus("ref.1", "Public Record(s)"), LegalStatus("ref.2", "Not Public Records"), LegalStatus("ref.3", "Public Records unless otherwise Stated"), LegalStatus("ref.4", "Welsh Public Record(s)")),
[info]     List(PlaceOfDeposit("1", "The National Archives, Kew"), PlaceOfDeposit("2", "British Museum, Department of Libraries and Archives"), PlaceOfDeposit("3", "British Library, National Sound Archive")),
[info]     List(Creator(uk.gov.nationalarchives.omega.editorial.models.Creator$CreatorType$CorporateBody$@2cd16f3f, "RR6", "100th (Gordon Highlanders) Regiment of Foot", Some(1794), Some(1894)), Creator(uk.gov.nationalarchives.omega.editorial.models.Creator$CreatorType$CorporateBody$@2cd16f3f, "JS8", "BBC", None, None), Creator(uk.gov.nationalarchives.omega.editorial.models.Creator$CreatorType$CorporateBody$@2cd16f3f, "S2", "The National Archives", Some(2003), None), Creator(uk.gov.nationalarchives.omega.editorial.models.Creator$CreatorType$Person$@4e9a5f95, "3FH", "Dainton, Sir Frederick Sydney", Some(1914), Some(1997)), Creator(uk.gov.nationalarchives.omega.editorial.models.Creator$CreatorType$Person$@4e9a5f95, "39K", "Cannon, John Francis Michael", Some(1930), None)),
[info]     Form(play.api.data.ObjectMapping16@36fbf5a2, HashMap("custodial-history" -> "", "end-date-month" -> "12", "covering-dates" -> "1969", "former-reference-department" -> "", "oci" -> "COAL.2022.V8RJW.P", "creator-ids[0]" -> "92W", "legal-status-id" -> "ref.1", "former-reference-pro" -> "", "note" -> "", "end-date-day" -> "31", "end-date-year" -> "1969", "creator-ids[1]" -> "48N", "action" -> "removeLastCreator", "start-date-month" -> "1", "start-date-year" -> "1969", "scope-and-content" -> "Bedlington Colliery, Newcastle Upon Tyne. Photograph depicting: view of pithead baths. (H)", "background" -> "", "place-of-deposit-id" -> "1", "start-date-day" -> "1"), List(), Some(EditSetRecordFormValues("Bedlington Colliery, Newcastle Upon Tyne. Photograph depicting: view of pithead baths. (H)", "1969", "", "", "1", "1", "1969", "31", "12", "1969", "ref.1", "1", "", "", "", List("92W", "48N", "8R6")))),
[info]     MessagesImpl(Lang(en_GB), play.api.i18n.DefaultMessagesApi@7dee08f8),
[info]     GET /
[info] );
[info]     -> at uk.gov.nationalarchives.omega.editorial.controllers.EditSetRecordController.generateEditSetRecordEditView(EditSetRecordController.scala:465)
[info]  - has following stubbing(s) with different arguments:
[info]     1. editSetRecordEdit.apply(
[info]     null,
[info]     null,
[info]     null,
[info]     null,
[info]     null,
[info]     null,
[info]     null,
[info]     null,
[info]     null,
[info]     null
[info] );
[info]       -> at controllers.EditSetRecordControllerSpec.givenEditViewIsGenerated(EditSetRecordControllerSpec.scala:1779)
[info] Typically, stubbing argument mismatch indicates user mistake when writing tests.
[info] Mockito fails early so that you can debug potential problem easily.
[info] However, there are legit scenarios when this exception generates false negative signal:
[info]   - stubbing the same method multiple times using 'given().will()' or 'when().then()' API
[info]     Please use 'will().given()' or 'doReturn().when()' API for stubbing.
[info]   - stubbed method is intentionally invoked with different arguments by code under test
[info]     Please use default or 'silent' JUnit Rule (equivalent of Strictness.LENIENT).
[info] For more information see javadoc for PotentialStubbingProblem class.
[info]         at uk.gov.nationalarchives.omega.editorial.controllers.EditSetRecordController.generateEditSetRecordEditView(EditSetRecordController.scala:465)
[info]         at uk.gov.nationalarchives.omega.editorial.controllers.EditSetRecordController.$anonfun$removeLastCreator$3(EditSetRecordController.scala:290)
[info]         at map @ uk.gov.nationalarchives.omega.editorial.controllers.EditSetRecordController.$anonfun$removeLastCreator$2(EditSetRecordController.scala:280)
[info]         at flatMap @ uk.gov.nationalarchives.omega.editorial.controllers.EditSetRecordController.$anonfun$removeLastCreator$1(EditSetRecordController.scala:279)
[info]         at flatMap @ uk.gov.nationalarchives.omega.editorial.controllers.EditSetRecordController.removeLastCreator(EditSetRecordController.scala:278)
[info]         at flatMap @ uk.gov.nationalarchives.omega.editorial.controllers.EditSetRecordController.submit(EditSetRecordController.scala:119)
[info]         at map @ uk.gov.nationalarchives.omega.editorial.controllers.BaseAppController.findEditSetRecord(BaseAppController.scala:53)
[info]         at map @ uk.gov.nationalarchives.omega.editorial.controllers.EditSetRecordController.$anonfun$generateResult$2(EditSetRecordController.scala:151)
[info]         at map @ uk.gov.nationalarchives.omega.editorial.controllers.BaseAppController.findEditSet(BaseAppController.scala:48)
[info]         at flatMap @ uk.gov.nationalarchives.omega.editorial.controllers.EditSetRecordController.$anonfun$generateResult$1(EditSetRecordController.scala:150)
[info]         ...

```
I've spent days debugging mockito errors before, so I'm going to come back to this one after fixing all the other bugs

Reply:
Confirmed.